### PR TITLE
[IMP] website: introduce `s_pricelist_cafe` snippet

### PR DESCRIPTION
**Pull request by Brieuc-brd**
This commit adds the new `s_pricelist_cafe` snippet.

task-4094408
Part of task-4077427

Capture d’écran 2024-08-09 à 15 48 21

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Brieuc-brd** [comment]
Hey @stefanorigano 👋 

I've finished to code this snippet. But it needs the merge of this one before :https://github.com/odoo/odoo/pull/176145 (so it's normal if this branch is red) I temporarily cherry picked its commit here.
What do you think ? 

Thanks in advance !

**Comment by Brieuc-brd** [comment]
@stefanorigano 👋 
This snippet is now fully independent and ready to be reviewed :) 
As the context is very specific, I didn't do any customization in the themes and let the color presets do the work, what do you think ?

File: addons/website/static/src/snippets/s_pricelist_cafe/options.js
Comment (stefanorigano) [inline addons/website/static/src/snippets/s_pricelist_cafe/options.js:16]: Nitpicking since we'll have to apply some changes anyways.
(don't forget to adapt wherever this variable it's used as well)

```suggestion
        const drinks = this.$target[0].querySelectorAll('.s_pricelist_cafe_item');
```

File: addons/website/views/snippets/s_pricelist_cafe.xml
Comment (stefanorigano) [inline addons/website/views/snippets/s_pricelist_cafe.xml:7]: Overall layout: this approach generate a suboptimal result with `.container`. The result with `.container-fluid` is good, but it's not a **real** "full-width result" since columns  appear to be constrained in an invisible container smaller than the default one 😉

| .container | .container-fluid |
|--------|--------|
|image| image | 

Proposal:
- Text columns: remove the `offset-lg-1` classes and switch  to `col-lg-5`
- Use `.container` by default 

| .container  | .container-fluid  |
|--------|--------|
| image | image | 

(OFC `.o_container_small` is doomed regardless, but we'll let the user decide if use it or not... ).

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Very good, thanks!
Just one layout issue (very easy to address), for the rest details...
...almost there, good job 🙌

File: addons/website/views/snippets/s_pricelist_cafe.xml
Comment (Brieuc-brd) [reply addons/website/views/snippets/s_pricelist_cafe.xml:7]: As we discussed, I've used `container-fluid`+ `offset-lg-1` to avoid this weird rendering that we have in medium breakpoint when we used `container` :
Capture d’écran 2024-08-12 à 14 31 22

IMO this is fine. If it's ok for you as well we can go with `container`, wdyt ?

Comment (stefanorigano) [reply addons/website/views/snippets/s_pricelist_cafe.xml]:
```scss
@include media-breakpoint-down(lg) {
    .s_pricelist_cafe > .o_web_editor_Origins_19:not(. o_shape_show_mobile) {
       display: none;
    }
}
```
Breakdown:
- Scenario 1: If the user retains both the shape and the configuration (which hides it on mobile), the behavior will extend to the `md` breakpoint as well—this is the expected result. 👌
- Scenario 2: If the user keeps the shape but opts to show it on mobile, this would be an intentional choice. While a visual glitch may occur, it's ultimately up to the user to decide if this is acceptable.
- Scenario 3: If the user changes or removes the shape, this rule won’t be triggered at all.

What do you think? If you agree, go ahead and apply the changes, but please keep this thread "unresolved" so the reviewer can easily understand the rationale behind the decision. This CSS block should eventually include proper comments for clarity.

> As we discussed, I've used con...

I've been reviewing over 10 PRs today, so apologies if I missed any details 😉

Comment (Brieuc-brd) [reply addons/website/views/snippets/s_pricelist_cafe.xml]: Looks nice 👍 I added the code, thanks !

**Comment by Brieuc-brd** [comment]
Thanks for your review @stefanorigano 🙏 
I updated the branch 👍

**Review by stefanorigano — APPROVED** [review]
LGTM @Brieuc-brd  👍 
Don't forget the comment in the CSS.

Thanks!

File: addons/website/static/src/snippets/s_pricelist_cafe/000.scss
Comment (Brieuc-brd) [reply addons/website/static/src/snippets/s_pricelist_cafe/000.scss]: W'oops 🙈 
Done ! ✅  https://github.com/odoo/odoo/compare/86674195afcaf1df2b63dd42e70615a8116dc06e..bafc924c20647adfcdf81da4bd8a2fc626c321e4

**Comment by Brieuc-brd** [comment]
Record naming fixed : https://github.com/odoo/odoo/compare/bafc924c20647adfcdf81da4bd8a2fc626c321e4..1a0d38a97178aa579c69cfdf20387a92eebabab7

**Comment by bso-odoo** [comment]
LGTM, queud after `s_faq_list` #175916

File: addons/website/views/snippets/s_pricelist_cafe.xml
Comment (SergeBayet) [inline addons/website/views/snippets/s_pricelist_cafe.xml:10]: This will break the display options
```suggestion
                    <h2>
                          <span class="h3-fs">Discover our drinks</span>
                    </h2>
```

File: addons/website/views/snippets/s_pricelist_cafe.xml
Comment (SergeBayet) [inline addons/website/views/snippets/s_pricelist_cafe.xml:100]: @Brieuc-brd When I test these options appear twice : 
image

**Review by SergeBayet — CHANGES_REQUESTED** [review]
@Brieuc-brd Great snippet. :+1:  A few comments however. See below.

File: addons/website/views/snippets/s_pricelist_cafe.xml
Comment (bso-odoo) [reply addons/website/views/snippets/s_pricelist_cafe.xml:100]: It seems this is the case only after rebase - additionally it does not work anymore, while it did before the rebase.

Comment (Brieuc-brd) [reply addons/website/views/snippets/s_pricelist_cafe.xml:100]: This seems to be a bug introduced in https://github.com/odoo/odoo/pull/176631 😬 
I've made a fix : https://github.com/odoo/odoo/pull/179098
@SergeBayet since you reviewed the initial PR, could you have a look at my fix ? :)

File: addons/website/views/snippets/s_pricelist_cafe.xml
Comment (Brieuc-brd) [reply addons/website/views/snippets/s_pricelist_cafe.xml]: Good catch, thanks 👍  It's fixed ✅  https://github.com/odoo/odoo/compare/c47fe89675bd4d09ea2fc48fbf5b7858c34395da..d8a0d5f5e64ef4a3ca844d7916ec0d25be07c65b

Comment (Brieuc-brd) [reply addons/website/views/snippets/s_pricelist_cafe.xml:10]: EDIT: as discussed, I removed the span https://github.com/odoo/odoo/compare/d8a0d5f5e64ef4a3ca844d7916ec0d25be07c65b..9103b43598fc4f995df375982e1cdb7d5ffb584c

**Comment by bso-odoo** [comment]
@robodoo delegate=bvr-odoo

**Review by SergeBayet — APPROVED** [review]

**Comment by bvr-odoo** [comment]
@robodoo r+

**Comment by seb-odoo** [comment]
Hello, it looks like this PR is causing issues in staging when including design-theme in the build:

https://runbot.odoo.com/runbot/build/68130892

Removing the PR from the queue until the issue can be resolved.

@robodoo r-

**Comment by qsm-odoo** [comment]
@Brieuc-brd If I am not mistaken it's because the test of @bso-odoo is a bit too violent: he does not allow multiple `s_` classes on elements except if you explicitely whitelist them in his test. @bso-odoo we should review that, using multiple `s_` classes for options is something I would recommend.

@Brieuc-brd Meanwhile, I think you can just remove the `s_pricelist_cafe_wrap_products` class from your snippet, it is unused. It should be ok to merge afterwards.

**Comment by Brieuc-brd** [comment]
Thanks for the info @qsm-odoo ! 🙏 
I removed the class and rebased the branch 👍

**Comment by bvr-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
